xDripG5/Transmitter.swift

```diff
@@ -280,6 +315,19 @@ fileprivate extension PeripheralManager {
 
         let activationDate = Date(timeIntervalSinceNow: -TimeInterval(timeMessage.currentTime))
 
+        while let message = getMessage() {
```

Comment:
These queued messages should run after the GlucoseRxMessage is received (and the delegate method called).
The client should consider the most up-to-date status of the sensor when choosing which additional messages to run, for example, a session start wouldn't be necessary if the sensor was already started, or the index of a new glucose value would inform whether backfill should be read.

Reply:
The official iOS app calls the queued messages before the GlucoseRxMessage is received (confirmed by watching the messages on the control characteristic while running xDripG5 in passive mode). One advantage of this is that when the glucose message is read it is immediately updated based on the queued messages. For example, a calibration will be immediately reflected in the glucose that is read; and after a sensor start message the state will immediately switch to `.warmup`. Otherwise one would have to wait 5 minutes for the next read.
I agree that the client should restrict which messages can be sent based on the previous state, but IMO we should just test for freshness of the last read glucose (perhaps < 15 mins old) and use the state contained in that message. In the event that a message is not compatible with the current state (e.g. a start message while in warmup) the transmitter just ignores it.